src/hooks/useAddLiquidityHandler.tsx

```diff
@@ -107,20 +110,18 @@ export const useAddLiquidityHandler = () => {
     fetchContract()
   }, [provider])
 
-  const getTokenBalanceAsync = async (setTokenBalance: Dispatch<SetStateAction<string>>, token: Token) => {
-    if (provider && token.address) {
-      let accountAddress = accountInfo?.address || ''
-      let balance = await getTokenBalance(accountAddress, token.address, token.symbol, provider)
-      setTokenBalance(balance || '0')
-    }
-  }
-
   useEffect(() => {
-    getTokenBalanceAsync(setTokenABalance, tokenASelected)
+    let accountAddress = accountInfo?.address || ''
+    if (provider && tokenASelected.address) {
+      getTokenBalance(accountAddress, tokenASelected.address, tokenASelected.symbol, provider, setTokenABalance)
```

Comment:
should be:
let bal = getTokenBalance()
setTokenBalance(bal)
reduce coupling, more reusable.

Reply:
Can not do as getTokenBalance is a promise so we will have to wait for it to be fulfilled. And we can not use await directly here